frappe/core/doctype/user/user.py

```diff
@@ -334,6 +340,10 @@ def on_trash(self):
 			set `user`=null
 			where `user`=%s""", (self.name))
 
+		# delete notification settings
+		frappe.db.sql("""DELETE FROM `tabNotification Settings` WHERE `name`=%s""", 
+			(self.name))
```

Comment:
Try and use the ORM for this one, otherwise it will bypass any `on_trash` controller events